tools/src/vendoring/config/expoGoConfig.ts

```diff
@@ -51,7 +51,23 @@ const config: VendoringTargetConfig = {
     'lottie-react-native': {
       source: 'lottie-react-native',
       sourceType: 'npm',
-      ios: {},
+      ios: {
+        async preReadPodspecHookAsync(podspecPath: string): Promise<string> {
+          let content = await fs.readFile(podspecPath, 'utf-8');
+          // load `install_modules_dependencies()` from react-native_pods.rb
+          content = content.replace(
+            /^(\s*install_modules_dependencies\(.*$)/gm,
+            `\
+    unless defined?(install_modules_dependencies)
+      require File.join('${REACT_NATIVE_SUBMODULE_DIR}', 'scripts/react_native_pods')
+    end\n$1`
+          );
+          // comment out `Pod::UI.puts` calls that will break JSON parsing
+          content = content.replace(/Pod::UI.puts\(.+\)/gm, '');
+          await fs.writeFile(podspecPath, content);
+          return podspecPath;
+        },
+      },
```

Comment:
Sadly, it's gonna be a nightmare. Meta proposal for the `install_modules_dependencies` method in the podspecs is an abuse of CocoaPods in my opinion. Specs should never depend on some external things, without importing them. We shouldn't assume that it's imported from the Podfile context.

Reply:
although the breaks cocoapods assumption, i think in the react-native ecosystem, we need a way to unify some fragmented settings like `HEADER_SEARCH_PATHS` or `folly_compiler_flags`. that's not ideal for 3rd party libs to maintain these settings by their own. if we were to add some unified settings inside podspecs, we would ultimately introduce something similar to the `install_modules_dependencies`, and then break cocoapods assumption.

that's not ideal but so far a more maintainable solution.